packages/types/src/codec/Set.ts

```diff
@@ -78,21 +78,34 @@ export default class Set extends Base<Array<string>> implements Codec {
    * @description true is the Set contains no values
    */
   get isEmpty (): boolean {
-    return this.values.length === 0;
+    return this.size === 0;
   }
 
   /**
    * @description The actual set values as a Array<string>
    */
-  get values (): Array<string> {
-    return this.raw;
+  get strings (): Array<string> {
+    return [...super.values()];
   }
 
   /**
    * @description The encoded value for the set members
    */
   get valueEncoded (): number {
-    return Set.encodeSet(this._setValues, this.raw);
+    return CodecSet.encodeSet(this._setValues, this.strings);
+  }
+
+  /**
+   * @description adds a value to the Set (extended to allow for validity checking)
+   */
+  add (key: string): this {
+    // we have the isUndefined(this._setValues) in here as well, add is used internally
+    // in the Set constructor (so it is undefined at this point, and should allow)
```

Comment:
outdated comment?

Reply:
Maybe just unclear. So basically the assert does this `assert(isUndefined(this._setValues) || ...` - this is for the constructor (Set calls `add` internally)

Reply:
Ah, understood